Extensions/DataSources/src/DataSourcesHub/action-creators/DataSourcesActionCreators.tsx

```diff
@@ -0,0 +1,195 @@
+import * as Reflux from 'reflux';
+import { DataSourcesActions } from '../actions/DataSourcesActions';
+import { ServiceEndpointRestClient, ServiceEndpointDetails, ServiceEndpointType, DataSource, DataSourceDetails, ResultTransformationDetails, ServiceEndpointRequest, ServiceEndpointRequestResult, OAuthConfigurationActionFilter } from 'azure-devops-extension-api/ServiceEndpoint';
+import { getClient } from 'azure-devops-extension-api/Common';
+import { DataSourceInfo,PayloadForExecuteServiceEndpointRequest,PayloadForUpdateDataSource, PayloadForUpdateDataSourceParameters, PayloadForSelectDataSource, ParamInfo } from '../states/DataSourcesExtensionState';
+
+export class DataSourcesActionCreators extends Reflux.Component{
+
+    public updateDataSourceParameters(name:string,value:string) {
+        var payload:PayloadForUpdateDataSourceParameters = {
+            name:name,
+            value:value,
+            parseError:null,
+            executeError:null,
+            result:null,
+        };       
+        DataSourcesActions.UpdateDataSourceParameters(payload)
+    }
+
+    public selectDataSource(itemtext: string, datasourceInfo: string) {
+        var Parameters:ParamInfo=this.extractParameters(datasourceInfo,null)
+        var payload:PayloadForSelectDataSource = {
```

Comment:
rename PayloadForSelectDataSource to SelectDataSourcePayload

why do we have parseError, executeError and result here?